### DISCUSSION TITLE
Shelly 1 L with Button

### DISCUSSION THREAD

**Discussion by Xamude** [arendst/Tasmota#14518] [Q&A] [answered]
Hello, 
i have a Sehlly 1L flashed with Tasmaota. 
The Problem is that the Shelly switches Power on without someone press a Button.
When i disconnect all Wires from SW 1 and SW 2 the Problem still exists.

This is the Log with only L and N connectet an NO SW 1/2:

> 15:59:01.906 MQT: Raum/Kueche/stat/RESULT = {"Button1":{"Action":"SINGLE"}}
15:59:04.875 MQT: Raum/Kueche/stat/RESULT = {"Button1":{"Action":"SINGLE"}}
15:59:05.976 MQT: Raum/Kueche/stat/RESULT = {"Button1":{"Action":"SINGLE"}}
15:59:29.974 MQT: Raum/Kueche/stat/RESULT = {"Button1":{"Action":"SINGLE"}}
15:59:34.278 MQT: Raum/Kueche/stat/RESULT = {"Button1":{"Action":"SINGLE"}}
15:59:38.367 MQT: Raum/Kueche/stat/RESULT = {"Button1":{"Action":"SINGLE"}}
15:59:44.157 MQT: Raum/Kueche/stat/RESULT = {"Button1":{"Action":"SINGLE"}}
15:59:45.612 MQT: Raum/Kueche/stat/RESULT = {"Button1":{"Action":"SINGLE"}}
15:59:48.453 MQT: Raum/Kueche/stat/RESULT = {"Button1":{"Action":"TRIPLE"}}
15:59:50.011 MQT: Raum/Kueche/stat/RESULT = {"Button1":{"Action":"SINGLE"}}
15:59:53.478 MQT: Raum/Kueche/stat/RESULT = {"Button1":{"Action":"SINGLE"}}

The Template i used:
`{"NAME":"Shelly 1L","GPIO":[320,0,0,0,128,224,0,0,0,0,129,0,0,4736],"FLAG":0,"BASE":18}`
128,129 = Button_in | Button inverted, active high no internal pull-up resistor
I use a similar  Template configuration on Shelly 1 without issues, so i think this is not the Problem.

I also flashed a second Shelly 1L there are the same Problem. 
Flashed one back to Stock FW to test, if use her the config Button there´s no "Ghost Switching"
(there is no Console at Stock to get more Information than the Light switches on)

Hope someone can help me here :)

**Comment by barbudor** [comment, marked as answer]
Proper recommended template is `{"NAME":"Shelly 1L","GPIO":[320,0,0,0,192,224,0,0,0,0,193,0,0,4736],"FLAG":0,"BASE":18}`
This is a 1 Relay device, you shouldn't have more than 1 Button defined.
Use SwitchMode 4 or SwitchMode 3 to use push-button on the switches input

**Reply by sfromis** [reply]
According to the wiring diagram from Alterco, the Shelly 1L does have two switch inputs, SW1 and SW2, with Sx providing signal for those.
![image](https://user-images.githubusercontent.com/47082390/150213919-f90a8d5c-0e9a-45ca-af95-da68e4cab2c4.png)

**Reply by Xamude** [reply]
yes its the original Template from [https://templates.blakadder.com/shelly_1L.html](url)
but its for switches and i replaced 192/193 with 128/129 für inv. Buttons.

i use setoption1 1 i ned button pres up to 4x 

when i can use only 1 Button not 2 then shelly 1L is useless :D i need to connect 2 Buttons:
Button1: connected to the Relay (Single Press) , Websend to Shelly 1 (Double Press)
Button2 is for Sehllys in a other Room: Websend to Shelly 1 (Single Press) , Websend to Shelly 1 (Double Press)

if know this trouble before i bought the 1L i definitively bought a 2.5 , this works fine at a other Room, but here i only ned 1 Relay so i bought 1L 🙈

will test at the weekend with other Template settings 

EDIT: 
@sfromis 
Yes but i have this Issues Without a Wire at SW 1 / SW 2 /Sx
First I thought of a wrong wiring, but Issue continues without connectet Wires

**Reply by sfromis** [reply]
Are you using it in no neutral mode? With the bypass in place? Using no neutral incurs a higher risk of trouble, if the wiring and power availability is not exactly right.

**Reply by Xamude** [reply]
Using it with Neutral.

**Reply by barbudor** [reply]
> but its for switches and i replaced 192/193 with 128/129 für inv. Buttons.

Which is wrong as I told you
Use Switches as intended and apply the proper Switchmode

**Reply by barbudor** [reply]
ok, you want to use multipress, that's why you use Buttons

I still suggest that you try 1st with the original template to see if you get ghost events

**Reply by Xamude** [reply]
Sorry for the late response, had not enough time to test last weekend.

If i use the original template i have not seen the "ghost switching"
Tested different settings.
What i can say at the moment it looks like a problem only on SW1:

SW1: Button_in // SW2: Switch_n = ghost switching
SW1: Switch_n // SW2: Button_in = no entries in the console

**Reply by ascillato** [reply]
Shelly uses specific code for managing its inputs. If you change the GPIO assignment from switch to button, Tasmota will not be using Shelly's specific code. Please, do as @barbudor have explained -> Use the default Template and modify the behavior of the input by command SWITCHMODE.

**Reply by Xamude** [reply]
Tested with standard Template and Switchmode 3 and 4 no "ghost switching"

**Reply by Xamude** [reply]
And now how to configure the Shelly right to use Multi-Press ?
Need Single to Quad Press.

**Reply by ascillato** [reply]
Switchmode 5 or 6 and set the configuration as you need following https://tasmota.github.io/docs/Buttons-and-Switches/#changing-default-functionality
